pageserver/src/billing_metrics.rs

```diff
@@ -51,13 +61,21 @@ impl BillingMetric {
         tenant_id: TenantId,
         timeline_id: Option<TimelineId>,
         value: u64,
+        node_id: NodeId,
     ) -> Self {
         Self {
             metric,
             metric_type: "absolute",
             tenant_id,
             timeline_id,
             time: Utc::now(),
+            // key that allows metric collector to distinguish unique events
+            idempotency_key: format!(
+                "{}-{}-{:04}",
+                Utc::now(),
+                node_id,
+                rand::thread_rng().gen_range(0..=9999)
```

Comment:
Note: `rand::thread_rng()` was mentioned to be costly (somewhat similar to Regex::new where it's compiled vs Regex::match on the compiled later).

We might want to reuse this rng, if possible.

Reply:
Taking current timestamp is also not that cheap :)

Reply:
Yep, I see nothing critical here and don't ask for changes too :)

Still, `Utc::now` call is inevitable, while `rand::thread_rng` is not and that's what I wanted to notice.

Reply:
Please, check the last commit.
It is still pretty limited, though, because it's impossible to reuse thread_rng() with async code (or at least I don't know how to do it).